cmd/insprd/memory/tree/brokers.go

```diff
@@ -0,0 +1,60 @@
+package tree
+
+import (
+	"github.com/inspr/inspr/cmd/insprd/memory"
+	"github.com/inspr/inspr/pkg/ierrors"
+	"github.com/inspr/inspr/pkg/meta/brokers"
+	metautils "github.com/inspr/inspr/pkg/meta/utils"
+	"github.com/inspr/inspr/pkg/utils"
+)
+
+// BrokersMemoryManager implements the methods described by the BrokersInterface
+type BrokersMemoryManager struct {
+}
+
+// Brokers is a MemoryManager method that provides an access point for Alias
+func (tmm *MemoryManager) Brokers() memory.BrokerInterface {
+	return &BrokersMemoryManager{}
+}
+
+var bro *brokers.Brokers
+
+// GetAll returns an array containing all currently configured brokers
+func (bmm *BrokersMemoryManager) GetAll() utils.StringArray {
+	return bmm.get().Availible.ToArray()
+}
+
+// GetDefault returns the broker configured as default
+func (bmm *BrokersMemoryManager) GetDefault() string {
+	return bmm.get().Default
+}
+
+func (bmm *BrokersMemoryManager) get() *brokers.Brokers {
+	if bro == nil {
+		bro = &brokers.Brokers{
+			Availible: make(metautils.StrSet),
+		}
+	}
+	return bro
+}
+
+// Create configures a new broker on insprd
+func (bmm *BrokersMemoryManager) Create(broker string, config interface{}) error {
+	if ok := bmm.get().Availible[broker]; ok {
+		return ierrors.NewError().Message("error: %s is already configured on memory", broker).Build()
+	}
+	//configure the sidecarFactory for the given broker
```

Comment:
Just to make sure that its making sense for me, this wouldn't be much of "configure the sidecarFactory", but "select the right sidecarFactory for given broker and run it" instead, right?

Reply:
No, it would be configure the sidecar factory with a new broker